src/pybind11.hpp

```diff
@@ -90,6 +90,10 @@ class PyObjectGeography : public py::object {
         return py::cast(std::move(geog_ptr));
     }
 
+    // static PyObjectGeography as_py_object(std::unique_ptr<T> geog_ptr) {
+    //     return static_cast<PyObjectGeography>(py::cast(std::move(geog_ptr)));
+    // }
```

Comment:
I need to clean this up, but I thought to have `as_py_object` actually return PyObjectGeography instead of py::object, since it's currently not yet used, and in practice we need PyObjectGeography in the vectorized functions. But didn't get this to work ..

Reply:
Something like this should work:

```cpp
template <class T, std::enable_if<...>>
static PyObjectGeography as_py_object(std::unique_ptr<T> geog_ptr) {
    auto pyobj = py::cast(std::move(geog_ptr));
    auto pyobj_geog = static_cast<PyObjectGeography&>(pyobj);
    return std::move(pyobj_geog);
}
```

Reply:
Thanks, will give that a try. Are you OK with changing the existing as_py_object to do that? (I am not sure if we have a use case of the current version that returns py::object)

Reply:
Yes sure, it is also more consistent to have a static method returning an instance of the same type IMO. (`from_geog` could be a better name than `as_py_object`).